x/ccv/provider/keeper/key_assignment.go

```diff
@@ -128,12 +130,16 @@ func (k Keeper) GetValidatorByConsumerAddr(
 func (k Keeper) SetValidatorByConsumerAddr(
 	ctx sdk.Context,
 	chainID string,
-	consumerAddr sdk.ConsAddress,
-	providerAddr sdk.ConsAddress,
+	consumerAddr types.ConsumerConsAddress,
+	providerAddr types.ProviderConsAddress,
 ) {
 	store := ctx.KVStore(k.storeKey)
 	// Cons address is a type alias for a byte string, no marshaling needed
-	bz := providerAddr
+	bz, err := providerAddr.Marshal()
+	if err != nil {
```

Comment:
Technically a change in behavior, however an invalid cons address getting persisted would eventually cause critical errors. Addresses should be validated by this point (correct me if I'm wrong)

See `TestValidatorByConsumerAddrCRUD`